src/main/java/util/LocalizationUtil.java

```diff
@@ -34,4 +37,24 @@ public static final Date stringtoDateWithTimeConverter(String string) throws Par
             return null;
         return dateFormat.parse(string);
     }
+
+
+    public static final Date getWeekStartingDate(Date date) throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(DATE);
+        Calendar calendar=Calendar.getInstance();
```

Comment:
Make sure to set time as 00 to cover the whole date.

Reply:
ya the time is 00 in all starting date